### ISSUE TITLE
Workspace trust - update issue reporter

### ISSUE THREAD

**Issue by lszomoru** [microsoft/vscode#124337]
- [ ] Include workspace trust state when filing an issue through the Issue reporter
- [ ] Update the GitHub bug template so that the user tries to reproduce the issue in a trusted workspace

**Comment by bpasero** [comment]
I believe we should not allow a user to file an issue in restricted mode and explicitly ask to retry in a trusted workspace.

**Comment by TylerLeonhardt** [comment]
But what if they're trying to report an issue with restricted mode? Guide them to file it on GitHub directly?

**Comment by bpasero** [comment]
Guide the user into how to trust the workspace maybe?

**Comment by TylerLeonhardt** [comment]
I guess what I'm trying to say is, what if they wanna open an issue _about_ untrusted workspaces? And what about the extension issue filing? What if they wanna open an issue saying "your extension executes something out of the workspace in an untrusted workspace"?

**Comment by bpasero** [comment]
Yeah good point. I think my point is just this: once we enable workspace trust there is a high likely hood that we get issues from users that now see functionality not working as before. If on the way of reporting an issue we can make that clear to the user and then maybe avoid having the user filing an issue that would be ideal.

Imho worse would be to just have a boolean somewhere in the meta portion of the issue saying that the workspace is not trusted because then we still get a flood of issues coming in that we could have prevented.

**Comment by TylerLeonhardt** [comment]
Yeah I totally agree. How about this:

1. We disable filing an issue on VS Code if in an untrusted workspace
2. For extensions, we add a warning either above the submit button, or a model after clicking the submit button that says "Your workspace is untrusted so you're already experiencing limited functionality." (need better words)
3. We add another section in the drop down for "Workspace Trust" to just be about workspace trust?

Or we can add the warning everywhere... I can see folks opening issues on the "marketplace" because their extension they just installed "isn't loading" because of WT.

**Comment by sbatten** [comment]
For extensions, we know exactly what extension they are filing the issue for and whether or not the extension has limitations in restricted mode. We should be smart and add these kind of checks before alerting or limiting the user.

**Comment by TylerLeonhardt** [comment]
I have some stuff around Auth I need to fix this iteration but I plan on bringing this up for July as there are a couple things I wanna fix in the issue reporter.
